modules/health-check/health-check-fs2/src/main/scala/higherkindness/mu/rpc/healthcheck/handler/HealthServiceImpl.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.rpc.healthcheck.handler
+
+import cats.effect.{Async, Concurrent, Sync}
+import cats.effect.concurrent.Ref
+import higherkindness.mu.rpc.healthcheck._
+import higherkindness.mu.rpc.healthcheck.service.HealthCheckServiceFS2
+
+import cats.implicits._
+import fs2.Stream
+import fs2.concurrent.Topic
+
+object HealthService {
+
+  def buildInstance[F[_]: Sync: Concurrent]: F[HealthCheckServiceFS2[F]] = {
```

Comment:
You don't need Sync + Concurrent here. Maybe you could just require `ConcurrentEffect`?

Reply:
Actually with `Concurrent` it's enough. I'll check all the effects. I think `Async` it's not always needed